beacon-chain/p2p/peers/score_bad_responses.go

```diff
@@ -0,0 +1,84 @@
+package peers
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// BadResponsesThreshold returns the maximum number of bad responses a peer can provide before it is considered bad.
+func (s *PeerScorer) BadResponsesThreshold() int {
+	return s.config.BadResponsesThreshold
+}
+
+// BadResponses obtains the number of bad responses we have received from the given remote peer.
+func (s *PeerScorer) BadResponses(pid peer.ID) (int, error) {
+	s.store.RLock()
+	defer s.store.RUnlock()
+	return s.badResponses(pid)
+}
+
+// badResponses is a lock-free version of BadResponses.
+func (s *PeerScorer) badResponses(pid peer.ID) (int, error) {
+	if peerData, ok := s.store.peers[pid]; ok {
+		return peerData.badResponsesCount, nil
+	}
+	return -1, ErrPeerUnknown
+}
+
+// IncrementBadResponses increments the number of bad responses we have received from the given remote peer.
+// If peer doesn't exist this method is no-op.
+func (s *PeerScorer) IncrementBadResponses(pid peer.ID) {
+	s.store.Lock()
+	defer s.store.Unlock()
+
+	if _, ok := s.store.peers[pid]; !ok {
+		s.store.peers[pid] = &peerData{
+			badResponsesCount: 1,
+		}
+		return
+	}
+	s.store.peers[pid].badResponsesCount++
+}
+
+// IsBadPeer states if the peer is to be considered bad.
+// If the peer is unknown this will return `false`, which makes using this function easier than returning an error.
+func (s *PeerScorer) IsBadPeer(pid peer.ID) bool {
+	s.store.RLock()
+	defer s.store.RUnlock()
+	return s.isBadPeer(pid)
+}
+
+// isBadPeer is lock-free version of IsBadPeer.
+func (s *PeerScorer) isBadPeer(pid peer.ID) bool {
+	if peerData, ok := s.store.peers[pid]; ok {
+		return peerData.badResponsesCount >= s.config.BadResponsesThreshold
+	}
+	return false
+}
+
+// BadPeers returns the peers that are bad.
+func (s *PeerScorer) BadPeers() []peer.ID {
+	s.store.RLock()
+	defer s.store.RUnlock()
+
+	badPeers := make([]peer.ID, 0)
+	for pid, peerData := range s.store.peers {
+		if peerData.badResponsesCount >= s.config.BadResponsesThreshold {
```

Comment:
can we use the method `isBadPeer` over here ?

Reply:
Yep, good catch. Fixing.

Reply:
Done!